packages/forms/src/deprecated/widgets/EnumerationWidget/EnumerationWidget.js

```diff
@@ -252,8 +253,10 @@ class EnumerationForm extends React.Component {
 		};
 	}
 
-	componentWillReceiveProps(nextProps) {
-		this.setState(prevState => ({ ...prevState, items: nextProps.formData }));
+	componentDidUpdate(prevProps) {
+		if (this.props.formData !== prevProps.formData) {
```

Comment:
formData is an array.
An array  may not change by reference but still have new items in it.
I believe we may lost re render here.

Reply:
This widget is from the deprecated forms and I don't want to change any behavior.
The previous behavior was componentWillReceiveProps, this function is generally called if the props passed to the component change. This is done by reference. 
If you modify the array keeping the same reference, first it's a bad practice as we should prefer immutability, then it will not work as expected, like before.
Additional support for mutable array is not needed as this is deprecated.